### PR TITLE
[SDK-2603] Expose custom caching options from Auth0 SPA SDK

### PR DESCRIPTION
**Pull request by stevehobbsdev**
### Description

This PR upgrades `@auth0/auth0-spa-js` to [v1.16.0](https://github.com/auth0/auth0-spa-js/releases/tag/v1.16.0) and exposes the new custom cache option.

### References

https://github.com/auth0/auth0-spa-js/releases/tag/v1.16.0
https://github.com/auth0/auth0-spa-js/pull/743

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

**Review by adamjmcgrath — APPROVED** [review]

File: __tests__/auth-provider.test.tsx [hunk 1]
```diff
@@ -350,6 +350,7 @@ describe('Auth0Provider', () => {
     act(() => {
       result.current.logout({ localOnly: true });
     });
+    await waitForNextUpdate();
```
Comment (stevehobbsdev) [inline __tests__/auth-provider.test.tsx:353]: @adamjmcgrath Can you confirm if this is the way to handle this in the test? As the logout function is now async, I concluded that we'd need to wait until the next tick before asserting (it fails without this line).

Comment (adamjmcgrath) [reply __tests__/auth-provider.test.tsx]: @stevehobbsdev `logout` needs to remain sync by default here as well - I added c8742512a6da7244da50902f05bcf9cea91dd314

**Review by adamjmcgrath — APPROVED** [review]

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]
need to return the promise from logout

**Review by adamjmcgrath — DISMISSED** [review]

**Review by adamjmcgrath — APPROVED** [review]

File: __tests__/auth-provider.test.tsx (hunk 1 above)
Comment (stevehobbsdev) [reply __tests__/auth-provider.test.tsx]: Thanks for taking care of it 👍🏻

**Comment by benhunterandrewrobertson** [comment]
Awesome work guys, been holding out for this one. Much appreciated!

**Pull request merged**